nixos/modules/virtualisation/qemu-vm.nix

```
@@ -74,6 +80,20 @@ let
 
   drivesCmdLine = drives: concatStringsSep " " (imap1 driveCmdline drives);
 
+  driveDeviceName = idx:
+    let letter = elemAt lowerChars idx;
+    in if cfg.qemu.diskInterface == "scsi" then
+      "/dev/sd${letter}"
+    else
+      "/dev/vd${letter}";
+
+  lookupDriveDeviceName = driveName: driveList:
+    (findSingle (drive: drive.name == driveName)
+    (throw "Drive ${driveName} not found")
+    (throw "Multilple drives named ${driveName}") driveList).device;
```

Please reformat so that it's clearer what the arguments are, e.g.

```suggestion
  lookupDriveDeviceName = driveName: driveList: (
    findSingle
      (drive: drive.name == driveName)
      (throw "Drive ${driveName} not found")
      (throw "Multilple drives named ${driveName}")
      driveList
    ).device;
```

or

```suggestion
  lookupDriveDeviceName = driveName: driveList:
    let 
      drive = 
        findSingle
          (drive: drive.name == driveName)
          (throw "Drive ${driveName} not found")
          (throw "Multilple drives named ${driveName}")
          driveList;
    in
      drive.device;
```

---

Re-formatted with `nixfmt`.